### DISCUSSION TITLE
Multiple juicefs containers in an S3 bucket

### DISCUSSION THREAD

**Discussion by brechmos** [juicedata/juicefs#1599] [Q&A] [answered]
I was wondering if there was a way to have multiple juicefs containers within one S3 bucket (I am actually on DigitalOcean, but I don't suspect that matters). 

I formatted a juicefs container in a bucket within the "s3://mystor" directory (yeah, copy and paste from the tutorial). I setup to use redis for this.

I then created another directory in the S3 bucket "s3://new-dir" (beside "mystor") and tried to format a container within that new directory. But I got back the error:

```
2022/03/17 11:42:28.314610 juicefs[245143] <FATAL>: format: cannot update format from {Name:mystor UUID:74b67afc-032d-48dd-8103-f9e19a0c83ff Storage:space Bucket:https://<bucket>.nyc3.digitaloceanspaces.com/new-dir AccessKey:<key> SecretKey: <secret> BlockSize:4096 Compression:none Shards:0 Partitions:0 Capacity:0 Inodes:0 EncryptKey: TrashDays:1 MetaVersion:1 MinClientVersion: MaxClientVersion:} to {Name:new-dir UUID:74b67afc-032d-48dd-8103-f9e19a0c83ff Storage:space Bucket:https://<bucket>.nyc3.digitaloceanspaces.com/new-dir AccessKey:<key> SecretKey:<secret> BlockSize:4096 Compression:none Shards:0 Partitions:0 Capacity:0 Inodes:0 EncryptKey: TrashDays:1 MetaVersion:1 MinClientVersion: MaxClientVersion:} [format.go:403]
```

Is it possible to have two directories side-by-side in an S3 bucket each of which is a different juicefs container?

**Comment by davies** [comment, marked as answer]
Yes, please use different Redis DB and name for it. You tried to overwrite the existing JuiceFS mystor and got the above error.
